pkg/storage/ingest/reader.go

```diff
@@ -395,27 +406,36 @@ func (r *PartitionReader) fetchLastCommittedOffsetWithRetries(ctx context.Contex
 }
 
 func (r *PartitionReader) fetchLastCommittedOffset(ctx context.Context) (int64, error) {
-	const endOffset = -1 // -1 is a special value for kafka that means "the last offset"
-
 	// We use an ephemeral client to fetch the offset and then create a new client with this offset.
 	// The reason for this is that changing the offset of an existing client requires to have used this client for fetching at least once.
 	// We don't want to do noop fetches just to warm up the client, so we create a new client instead.
-	cl, err := kgo.NewClient(kgo.SeedBrokers(r.kafkaCfg.Address))
+	cl, err := kgo.NewClient(commonKafkaClientOptions(r.kafkaCfg, r.metrics.kprom, r.logger)...)
```

Comment:
Note to reviewers: we were missing metrics, logs and not applying the config (e.g. timeouts). This fixes it.